common/components/Header/components/CustomNodeModal.tsx

```diff
@@ -343,6 +341,7 @@ class CustomNodeModal extends React.Component<Props, State> {
 
   private saveAndAdd = () => {
     const node = this.makeCustomNodeConfigFromState();
+    shepherd.useProvider('myccustom', node.id, makeProviderConfig({ network: node.network }), node);
```

Comment:
Shepherd handles these race conditions internally, useProvider is considered a synchronous method from the caller's view

Reply:
Sorry if I wasn't clear, I wasn't worried about asynchronosity, I just thought this was a little too much logic for a component to implement. If there's ever another different component or process that can add custom nodes (e.g. pulling in a node from the URL) they would have to implement this as well. I figured this might be better off in a saga.